### PR TITLE
migrate other ehcaches

### PR DESCRIPTION
**Pull request by rcpeters**

### CODE REVIEW

File: orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidProfileCacheManagerImpl.java [hunk 1]
```diff
@@ -155,11 +171,10 @@ public void put(OrcidProfile orcidProfile) {
 
     public void put(String orcid, OrcidProfile orcidProfile) {
         try {
-            synchronized (profileLockers.obtainLock(orcid)) {
-                profileCache.put(new Element(new OrcidCacheKey(orcid, releaseName), orcidProfile));
-            }
+            profileCache.acquireWriteLockOnKey(orcid);
```
Comment (amontenegro) [inline orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidProfileCacheManagerImpl.java]: Hey @rcpeters, i think here you should create the key _new OrcidCacheKey(orcid, releaseName)_ and acquire the lock on that key, not in the orcid id right?

File: orcid-core/src/main/java/org/orcid/core/manager/impl/ProfileEntityCacheManagerImpl.java [hunk 2]
```diff
@@ -81,11 +82,10 @@ public void put(ProfileEntity profileEntity) {
     }
     public void put(String orcid, ProfileEntity profile) {
         try {
-            synchronized (lockers.obtainLock(orcid)) {
-                profileCache.put(new Element(new OrcidCacheKey(orcid, releaseName), profile));
-            }
+            profileCache.acquireWriteLockOnKey(orcid);
```
Comment (amontenegro) [inline orcid-core/src/main/java/org/orcid/core/manager/impl/ProfileEntityCacheManagerImpl.java]: @rcpeters same here, create the key and acquire the lock on that key

File: orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidProfileCacheManagerImpl.java (hunk 1 above)
Comment (rcpeters) [inline orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidProfileCacheManagerImpl.java]: hum.....if we are replacing `profileLockers.obtainLock(orcid)) ` then no. If we are saying the original code is incorrect then yes.

File: orcid-core/src/main/java/org/orcid/core/manager/impl/ProfileEntityCacheManagerImpl.java (hunk 2 above)
Comment (rcpeters) [inline orcid-core/src/main/java/org/orcid/core/manager/impl/ProfileEntityCacheManagerImpl.java]: hum.....if we are replacing `lockers.obtainLock(orcid)) ` then no. If we are saying the original code is incorrect then yes.

**Pull request merged**